streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImpl.java

```diff
@@ -42,14 +42,15 @@
     final Metrics metrics;
     final String groupName;
     final Map<String, String> tags;
+    final Map<Sensor, Sensor> parentSensors;
 
     public StreamsMetricsImpl(Metrics metrics, String groupName,  Map<String, String> tags) {
         Objects.requireNonNull(metrics, "Metrics cannot be null");
 
         this.metrics = metrics;
         this.groupName = groupName;
         this.tags = tags;
-
+        this.parentSensors = new HashMap<>();
```

Comment:
I am not sure if it is needed: `Metrics.sensor` will recursively link parent sensor to its children; `Metrics.removeSensor` will recursively remove its children sensor as well.